src/app/inventory/store/catalyst.ts

```diff
@@ -1,18 +1,30 @@
-import { DestinyProfileRecordsComponent, DestinyRecordState } from 'bungie-api-ts/destiny2';
+import {
+  DestinyCharacterRecordsComponent,
+  DestinyProfileRecordsComponent,
+  DestinyRecordState,
+} from 'bungie-api-ts/destiny2';
 import exoticToCatalystRecordHash from 'data/d2/exotic-to-catalyst-record.json';
 import exoticsWithCatalysts from 'data/d2/exotics-with-catalysts';
+import _ from 'lodash';
 import { DimCatalyst } from '../item-types';
 
 export function buildCatalystInfo(
   itemHash: number,
-  profileRecords: DestinyProfileRecordsComponent | undefined
+  profileRecords: DestinyProfileRecordsComponent | undefined,
+  characterRecords: { [key: string]: DestinyCharacterRecordsComponent } | undefined
 ): DimCatalyst | undefined {
   if (!exoticsWithCatalysts.has(itemHash)) {
     return undefined;
   }
 
   const recordHash = exoticToCatalystRecordHash[itemHash];
-  const record = recordHash && profileRecords?.records[recordHash];
+  const record =
+    recordHash &&
+    (profileRecords?.records[recordHash] ??
+      (characterRecords &&
+        _.compact(
+          Object.values(characterRecords).map((records) => records.records[recordHash])
+        )[0]));
```

Comment:
I *think* this is a bit more efficient? I suppose most cases will hit the `profileRecords` branch though, so maybe it doesn't matter.

```suggestion
        Object.values(characterRecords).find((records) => records.records[recordHash])?.records[recordHash]));
```